packages/apps/plugins/plugin-thread/src/hooks/analytics.ts

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { useCallback } from 'react';
+
+import { ObservabilityAction } from '@braneframe/plugin-observability/meta';
+import { useIntentDispatcher } from '@dxos/app-framework';
+import { type ReactiveObject } from '@dxos/echo-schema';
+import { getSpace } from '@dxos/react-client/echo';
+import { useOnTransition } from '@dxos/react-ui';
+
+export const useAnalyticsCallback = (space: string | undefined, name: string, meta?: any) => {
+  const dispatch = useIntentDispatcher();
+
+  return useCallback(
+    (dynamicMeta?: any) => {
+      void dispatch({
+        action: ObservabilityAction.SEND_EVENT,
+        data: { name, ...meta, ...dynamicMeta, space },
+      });
+    },
+    [dispatch, name, meta, space],
+  );
+};
+
+export const useOnEditAnalytics = (message: ReactiveObject<any>, editing: boolean) => {
+  const space = getSpace(message);
+
+  const onEditAnalytics = useAnalyticsCallback(space?.id, 'threads.comment-edited', {
+    messageId: message.id,
+    messageLength: message.text.length,
```

Comment:
will this capture the initial length or the edited length? probably the initial?

Reply:
Pretty sure it's mutating the doc on keypress like our other editors do so it should be relatively up to date. Clicking done-editing is a bit of a formality right.